gpkit/model.py

```diff
@@ -544,6 +544,30 @@ def _latex(self, unused=None):
                                  for var, val in self.constants.items()]) +
                          ["\\end{array}"])
 
+    def interact(self,  fn_of_sol=None, ranges=None, **solveargs):
+        try:
+            from ipywidgets import interactive, FloatSlider
+        except ImportError:
+            from IPython.html.widgets import interactive, FloatSliderWidget
+            FloatSlider = FloatSliderWidget
+
+        if ranges is None:
+            ranges = {k._cmpstr: FloatSlider(min=v/10.0, max=10*v,
+                                             step=v/10.0, value=v)
+                      for k, v in self.constants.items()}
+        if fn_of_sol is None:
+            def fn_of_sol(solution):
+                print solution.table(["cost", "freevariables"])
```

Comment:
Yeah. I just don't want it to get too long...maybe count the variables and choose that way? (prioritize free over senss or the other way around?)